mmdet/apis/det_inferencer.py

```diff
@@ -0,0 +1,472 @@
+        if pred_out_file != '':
+            mmengine.dump(result_dict, pred_out_file)
+        result_dict['visualization'] = visualization
+        return result_dict
+
+    def pred2dict(self,
+                  data_sample: DetDataSample,
+                  pred_out_file: str = '') -> Dict:
+        """Extract elements necessary to represent a prediction into a
+        dictionary.
+
+        It's better to contain only basic data elements such as strings and
+        numbers in order to guarantee it's json-serializable.
+
+        Args:
+            data_sample (:obj:`DetDataSample`): Predictions of the model.
+            pred_out_file: File to save the inference results w/o
+                visualization. If left as empty, no file will be saved.
+                Defaults to ''.
+
+        Returns:
+            dict: Prediction results.
+        """
+
+        result = {}
+        if 'pred_instances' in data_sample:
+            masks = data_sample.pred_instances.get('masks')
+            pred_instances = data_sample.pred_instances.numpy()
+            result = {
+                'bboxes': pred_instances.bboxes.tolist(),
+                'labels': pred_instances.labels.tolist(),
+                'scores': pred_instances.scores.tolist()
+            }
+            if masks is not None:
+                if pred_instances.bboxes.sum() == 0:
+                    # Fake bbox, such as the SOLO.
+                    bboxes = mask2bbox(masks.cpu()).numpy().tolist()
+                    result['bboxes'] = bboxes
+                encode_masks = encode_mask_results(pred_instances.masks)
+                for encode_mask in encode_masks:
+                    if isinstance(encode_mask['counts'], bytes):
+                        encode_mask['counts'] = encode_mask['counts'].decode()
+                result['masks'] = encode_masks
+
+        if 'pred_panoptic_seg' in data_sample and pred_out_file != '':
+            if VOID is None:
+                raise RuntimeError(
+                    'panopticapi is not installed, please install it by: '
+                    'pip install git+https://github.com/cocodataset/'
+                    'panopticapi.git.')
+
+            pan = data_sample.pred_panoptic_seg.sem_seg.cpu().numpy()[0]
+            pan[pan % INSTANCE_OFFSET == len(
+                self.model.dataset_meta['classes'])] = VOID
+            pan = id2rgb(pan).astype(np.uint8)
+
+            if 'img_path' in data_sample:
+                img_path = osp.basename(data_sample.img_path)
+                out_path = osp.splitext(img_path)[0] + '.png'
+                out_path = osp.join(osp.dirname(pred_out_file), out_path)
+            else:
+                out_path = osp.join(
+                    osp.dirname(pred_out_file),
+                    f'{self.num_predicted_imgs}.png')
+                self.num_predicted_imgs += 1
+
+            mmcv.imwrite(pan[:, :, ::-1], out_path)
+            result['panoptic_seg_path'] = out_path
+
+        return result
```

Comment:
Can this json file be visualized by the Visualizer?

Reply:
I notice that DetLocalVisulizer does not support RLE mask.

Reply:
The visualization in json format needs to add a script to support.